includes/gateways/paypal/paypal-standard.php

```diff
@@ -314,53 +314,19 @@ function give_process_paypal_web_accept( $data, $payment_id ) {
 	}
 
 	// Collect donation payment details.
-	$paypal_amount  = $data['mc_gross'];
+	$donation       = new Give_Payment( $payment_id );
 	$payment_status = strtolower( $data['payment_status'] );
-	$currency_code  = strtolower( $data['mc_currency'] );
-	$business_email = isset( $data['business'] ) && is_email( $data['business'] ) ? trim( $data['business'] ) : trim( $data['receiver_email'] );
-	$payment_meta   = give_get_payment_meta( $payment_id );
 
 	// Must be a PayPal standard IPN.
-	if ( 'paypal' !== give_get_payment_gateway( $payment_id ) ) {
-		return;
-	}
-
-	// Verify payment recipient.
-	if ( strcasecmp( $business_email, trim( give_get_option( 'paypal_email' ) ) ) !== 0 ) {
-
-		give_record_gateway_error(
-			__( 'IPN Error', 'give' ),
-			sprintf( /* translators: %s: Paypal IPN response */
-				__( 'Invalid business email in IPN response. IPN data: %s', 'give' ),
-				json_encode( $data )
-			),
-			$payment_id
-		);
-		give_update_payment_status( $payment_id, 'failed' );
-		give_insert_payment_note( $payment_id, __( 'Payment failed due to invalid PayPal business email.', 'give' ) );
-
-		return;
-	}
-
-	// Verify payment currency.
-	if ( $currency_code !== strtolower( $payment_meta['currency'] ) ) {
-
-		give_record_gateway_error(
-			__( 'IPN Error', 'give' ),
-			sprintf( /* translators: %s: Paypal IPN response */
-				__( 'Invalid currency in IPN response. IPN data: %s', 'give' ),
-				json_encode( $data )
-			),
-			$payment_id
-		);
-		give_update_payment_status( $payment_id, 'failed' );
-		give_insert_payment_note( $payment_id, __( 'Payment failed due to invalid currency in PayPal IPN.', 'give' ) );
-
+	// Validate donation id.
+	if (
+		! $donation->ID ||
+		'paypal' !== $donation->gateway ) {
 		return;
 	}
```

Comment:
Hmm, so does the `$payment_id` come from a payment that's already been associated with the PayPal transaction ID? Where are we validating that GiveWP has a payment with that corresponding PayPal transaction ID?